src/oauth2c_token_cache.erl

```diff
@@ -128,10 +127,10 @@ get_token(Key, ExpiryTimeLowerLimit) ->
       {error, not_found}
   end.
 
-get_expires_in(undefined, DefaultTTL) ->
+get_expire_time(undefined, DefaultTTL) ->
   erlang:system_time(second) + DefaultTTL;
-get_expires_in(ExpiryTime, _DefaultTTL) ->
-  ExpiryTime.
+get_expire_time(ExpireTime, _DefaultTTL) ->
+  ExpireTime.
```

Comment:
Isn't ExpireTime a relative value? Shouldn't this function return erlang:system_time(second) + ExpireTime to get an absolute expire time?

Reply:
So I reinstated the ExpireTime as the absolute value when the token expires. The change is that after we do the request to get the token we immediately calculate the ExpireTime and discard the expires_in value. At least that's what I'm trying to do. The code is complicated unfortunately.

Reply:
I missed that you are making the time absolute in the `do_retrieve_access_token ` function. Then everything is in order.